metrics_proto_api_test.go

```diff
@@ -405,12 +534,15 @@ func createFakeServer(t *testing.T) (*fakeMetricsServer, string, func()) {
 	go func() {
 		_ = srv.Serve(ln)
 	}()
+	_, serverPortStr, _ := net.SplitHostPort(ln.Addr().String())
+	conn := createConn(t, "localhost:"+serverPortStr)
+
 	stop := func() {
 		srv.Stop()
 		_ = ln.Close()
+		conn.Close()
```

Comment:
Nit: I think `conn` needs to be closed first.

Reply:
yes. fixed it.